src/prefect/experimental/workers/process.py

```diff
@@ -38,12 +46,24 @@ def _infrastructure_pid_from_process(process: anyio.abc.Process) -> str:
     return f"{hostname}:{process.pid}"
 
 
+class ProcessJobConfiguration(BaseJobConfiguration):
+    stream_output: bool = Field(template="{{ stream_output }}")
+    working_dir: Optional[Union[str, Path]] = Field(template="{{ working_dir }}")
+
+
+class ProcessVariables(BaseVariables):
+    stream_output: Optional[bool] = True
+    working_dir: Optional[Union[str, Path]] = None
```

Comment:
I think we should only type this a path. AFAIK, users can still provide a string and pydantic will validate strings passed in to make sure that they'er valid paths.
```suggestion
    working_dir: Optional[Union[Path]] = None
```

Reply:
I'm not sure Pydantic casts strings to path without a validator. You also need to be really careful with cross-platform support here i.e. win32/posix clients.

Reply:
The way it is currently mirrors the way it exists in the process infra block at the moment.

Reply:
According to the Pydantic docs, fields that are typed as a `Path` will cast strings to a `Path` without an explicit validator. I'm not 100% sure how this would work with different platforms, but I think it would use the path type that corresponds to the platform current process is running on since it just passes the value to `Path(v)`.

The reason I bring this up is because it will affect `jsonschema`'s ability to validate input server side. It might be worth investigating `jsonschema`'s ability to validate paths to make a decision. I don't think that needs to hold up this PR though, and can be addressed later if needed.